src/Voron/Data/VoronStream.cs

```diff
@@ -31,15 +32,22 @@ public VoronStream(Slice name, Tree.ChunkDetails[] chunksDetails, LowLevelTransa
             Name = name;
 
             _chunksDetails = chunksDetails;
-            _positions = new int[_chunksDetails.Length];
+            _chunksOffsets = new long[_chunksDetails.Length];
+
             _index = 0;
             _llt = llt;
             _lastPage = default(Page);
+            _position = 0;
 
-            foreach (var cd in _chunksDetails)
+            for (int index = 0; index < _chunksDetails.Length; index++)
             {
+                ref Tree.ChunkDetails cd = ref _chunksDetails[index];
                 Length += cd.ChunkSize;
+                _chunksOffsets[index] = _position;
+                _position += cd.ChunkSize;
```

Comment:
if we set it to zero immediately afterward, why set it here?

Reply:
I can create `tempPosition` here. I've to track global position.

Reply:
This is why I zero it immediately afterward.

Reply:
But... isn't that `Length` ?

Reply:
No, the position is global, so it is <0, Sum(chunk.Length))

To perform a binary search over it I've to persist the global position of each chunk, which means we've to calculate the sum of the length of all chunks before the current and persist it. After calculations, we're moving position to begin of the stream (which is 0)